--- a/src/values/types/string/interpolation.rs	2026-09-30 15:42:15.165004920 -0700
+++ b/src/values/types/string/interpolation.rs	2026-09-30 15:42:15.209005625 -0700
@@ -220,7 +220,7 @@
             Some(PercentSFormat::Str) => {
                 let arg = next_value()?;
                 match arg.unpack_str() {
-                    None => arg.collect_repr(&mut res),
+                    None => arg.collect_str(&mut res),
                     Some(s) => res.push_str(s),
                 }
             }
